File: src/Entity/Reservation.php
<?php

namespace App\Entity;

use App\Repository\ReservationRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;


#[ORM\Entity(repositoryClass: ReservationRepository::class)]
class Reservation
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    #[Assert\NotBlank(message:"The start date cannot be blank.")]
    #[Assert\GreaterThan(
         value:"today",
         message:"The start date must be after or on today")]
   
    private ?\DateTimeInterface $DateDebutR = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    #[Assert\NotBlank(message:"The end date cannot be blank.")]
    #[Assert\Expression(
        "this.getDateFinR() >= this.getDateDebutR()",
        message:"The end date must be after the start date."
    )]
    
    private ?\DateTimeInterface $DateFinR = null;

    #[ORM\Column]
    #[Assert\NotBlank(message:"The number of people attending cannot be blank.")]
     #[Assert\Type(
        type: 'integer',
        message: 'The value {{ value }} is not a valid {{ type }}.',
    )]
    private ?int $NbrPerso = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message:"Write room number and type : single,double or triple")]
    private ?string $TypeRoom = null;

    #[ORM\ManyToOne(inversedBy: 'reservation')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Hotel $IdHotel = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message:"You ID attending cannot be blank.")]
     #[Assert\Regex(
        pattern:"/^\d+$/",
        message:"Only numbers are allowed"
    )]
    private ?string $IdClent = null;

    #[ORM\OneToMany(mappedBy: 'reservation', targetEntity: User::class)]
    private Collection $Client;

    public function __construct()
    {
        $this->Client = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getDateDebutR(): ?\DateTimeInterface
    {
        return $this->DateDebutR;
    }

    public function setDateDebutR(\DateTimeInterface $DateDebutR): static
    {
        $this->DateDebutR = $DateDebutR;

        return $this;
    }

    public function getDateFinR(): ?\DateTimeInterface
    {
        return $this->DateFinR;
    }

    public function setDateFinR(\DateTimeInterface $DateFinR): static
    {
        $this->DateFinR = $DateFinR;

        return $this;
    }

    public function getNbrPerso(): ?int
    {
        return $this->NbrPerso;
    }

    public function setNbrPerso(int $NbrPerso): static
    {
        $this->NbrPerso = $NbrPerso;

        return $this;
    }

    public function getTypeRoom(): ?string
    {
        return $this->TypeRoom;
    }

    public function setTypeRoom(string $TypeRoom): static
    {
        $this->TypeRoom = $TypeRoom;

        return $this;
    }

    public function getIdHotel(): ?Hotel
    {
        return $this->IdHotel;
    }

    public function setIdHotel(?Hotel $IdHotel): static
    {
        $this->IdHotel = $IdHotel;

        return $this;
    }

    public function getIdClent(): ?string
    {
        return $this->IdClent;
    }

    public function setIdClent(string $IdClent): static
    {
        $this->IdClent = $IdClent;

        return $this;
    }

    /**
     * @return Collection<int, User>
     */
    public function getClient(): Collection
    {
        return $this->Client;
    }

    public function addClient(User $client): static
    {
        if (!$this->Client->contains($client)) {
            $this->Client->add($client);
            $client->setReservation($this);
        }

        return $this;
    }

    public function removeClient(User $client): static
    {
        if ($this->Client->removeElement($client)) {
            // set the owning side to null (unless already changed)
            if ($client->getReservation() === $this) {
                $client->setReservation(null);
            }
        }

        return $this;
    }
}

File: src/main/java/entities/Reservation.java
package entities;

import java.util.Date;

public class Reservation {
    private int id;
    private int id_hotel;

    public Reservation(int id, int id_hotel, int id_client, Date date_debut_r, Date date_fin_r, int nbr_perso, String type_room) {
        this.id = id;
        this.id_hotel = id_hotel;
        this.id_client = id_client;
        this.date_debut_r = date_debut_r;
        this.date_fin_r = date_fin_r;
        this.nbr_perso = nbr_perso;
        this.type_room = type_room;
    }

    private int id_client;
    private Date date_debut_r;
    private Date date_fin_r;
    private int nbr_perso;
    private String type_room;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getId_hotel() {
        return id_hotel;
    }

    public void setId_hotel(int id_hotel) {
        this.id_hotel = id_hotel;
    }

    public int getId_client() {
        return id_client;
    }

    public void setId_client(int id_client) {
        this.id_client = id_client;
    }

    public Date getDate_debut_r() {
        return date_debut_r;
    }

    public void setDate_debut_r(Date date_debut_r) {
        this.date_debut_r = date_debut_r;
    }

    public Date getDate_fin_r() {
        return date_fin_r;
    }

    public void setDate_fin_r(Date date_fin_r) {
        this.date_fin_r = date_fin_r;
    }

    public int getNbr_perso() {
        return nbr_perso;
    }

    public void setNbr_perso(int nbr_perso) {
        this.nbr_perso = nbr_perso;
    }

    public String getType_room() {
        return type_room;
    }

    public void setType_room(String type_room) {
        this.type_room = type_room;
    }




}
